runtime/compare.js

```diff
@@ -235,25 +235,25 @@ function caml_compare_val (a, b, total) {
 //Provides: caml_compare (const, const)
 //Requires: caml_compare_val
 function caml_compare (a, b) { return caml_compare_val (a, b, true); }
-//Provides: caml_int_compare mutable (const, const)
+//Provides: caml_int_compare (const, const)
```

Comment:
I'm not sure we should make this change. We are preventing some optimizations just to get an error in some bogus code that does not even use the result of the comparison.

Reply:
ok, I'll revert

Reply:
done